plant/plants/api/views.py

```diff
@@ -28,6 +44,47 @@ def get_queryset(self):
     def perform_create(self, serializer):
         serializer.save(owner=self.request.user)
 
+    @action(detail=True, methods=["GET"])
+    def plant_stats(self, request, pk=None):
+        start_date = request.query_params.get("start_date")
```

Comment:
I like to add another layer of validation for query parameters. In your case it's easy to pass some random value that passes if statement, but is definitely not valid for range filter on the database level and endpoint will throw 500.

I would add some serializer named for example StatsParamsSerializer and also assign default values (timezone.now() - timedelta(days=30) and timezone.now() so you also don't need if statement in L63
```
params = StatsParamsSerializer(request.query_params)
params.is_valid(raise_exception=True)
waterings_count = plant.waterings.filter(
                watering_date__range=(params.validated_data["start_date"], params.validated_data["end_date"])
            ).count()
```